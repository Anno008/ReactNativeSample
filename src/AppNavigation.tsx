import React from "react";
// Eventually try out react-native-navigation, better performance and richer configuration
// requires setup
import {
  createAppContainer,
  createBottomTabNavigator,
  NavigationContainer,
  TabScene,
} from "react-navigation";

import { HomeIcon } from "components/icons/HomeIcon";
import { InfoIcon } from "components/icons/InfoIcon";
import { getMainNavigationIcon } from "decorators/GetMainNavigationIcon";
import AboutScreen from "screens/AboutScreen";
import HomeScreen from "screens/HomeScreen";

const AppNavigation: NavigationContainer = createBottomTabNavigator(
  {
    About: {
      navigationOptions: {
        tabBarIcon: (props: TabScene) => getMainNavigationIcon(InfoIcon, props),
        tabBarLabel: "About"
      },
      screen: AboutScreen
    },
    Home: {
      navigationOptions: {
        tabBarIcon: (props: TabScene) => getMainNavigationIcon(HomeIcon, props),
        tabBarLabel: "Home"
      },
      screen: HomeScreen
    }
  },
  // Router config
  {
    initialRouteName: "Home",
    order: ["Home", "About"],
    tabBarOptions: {
      // activeBackgroundColor: "rgba(107, 185, 240, 0.6)",
      activeTintColor: "navy",
      inactiveTintColor: "gray",
      labelStyle: {
        fontSize: 14
      },
      style: {
        backgroundColor: "rgba(107, 185, 240, 0.5)",
        height: 50
      }
    }
  }
);

export default createAppContainer(AppNavigation);
