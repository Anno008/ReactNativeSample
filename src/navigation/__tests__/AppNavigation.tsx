import AppNavigation from "navigation/AppNavigation";
import { shallow } from "enzyme";
import React from "react";

describe("AppNavigation tests", (): void => {
    it("Should match AppNavigation snapshot", (): void => {
        const wrapper = shallow(<AppNavigation />);
        expect(wrapper).toMatchSnapshot();
    });
});
