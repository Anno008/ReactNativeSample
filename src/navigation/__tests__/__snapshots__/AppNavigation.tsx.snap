// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation tests Should match AppNavigation snapshot 1`] = `ShallowWrapper {}`;
